partner_preview_scripts/encore_parser.py

```diff
@@ -0,0 +1,528 @@
+    It has to be done when joining with other datasets.
+    !!!
+
+    Args:
+        cellPassportFile: Path to the cell passport file.
+    """
+
+    def __init__(self, cellPassportFile: str, spark) -> None:
+        self.cellPassportFile = cellPassportFile
+        self.spark = spark
+        self.cellMap = self.generate_map()
+
+    def generate_map(self) -> None:
+        """Reading and procesing cell line data from the cell passport file.
+
+        The schema of the returned dataframe is:
+
+        root
+        |-- name: string (nullable = true)
+        |-- id: string (nullable = true)
+        |-- biomarkerList: array (nullable = true)
+        |    |-- element: struct (containsNull = true)
+        |    |    |-- name: string (nullable = true)
+        |    |    |-- description: string (nullable = true)
+        |-- diseaseCellLines: struct (nullable = false)
+        |    |-- tissue: string (nullable = true)
+        |    |-- name: string (nullable = true)
+        |    |-- id: string (nullable = true)
+        |    |-- tissueId: string (nullable = true)
+
+        Note:
+            * Microsatellite stability is the only inferred biomarker.
+            * The cell line name has the dashes removed.
+            * Id is the cell line identifier from Sanger
+            * Tissue id is the UBERON identifier for the tissue, fetched from OLS
+        """
+
+        # loading cell line annotation data from Sanger:
+        cell_df = (
+            self.spark.read
+            .option("multiline", True) # <- this is crazy! Some annotation within the csv fields have newlines!
+            .csv(self.cellPassportFile, header=True, sep=',', quote='"')
+            .withColumn('biomarkerList', self.parse_msi_status(col('msi_status')))
+            .select(
+                col('model_name').alias('name'),
+                col('model_id').alias('id'),
+                col('tissue'),
+                col('biomarkerList')
+            )
+            .persist()
+        )
+
+        # Generating a unique set of tissues in a pandas dataframe:
+        tissues = cell_df.select('tissue').distinct().toPandas()
+        logging.info(f'Found {len(tissues)} tissues.')
+
+        # Generating a unique set of cell lines in a pandas series:
+        mappedTissues = tissues.assign(tissueId=lambda df: df.tissue.apply(self.lookup_uberon))
+        logging.info(f'Found mapping for {len(mappedTissues.loc[mappedTissues.tissueId.notna()])} tissues.')
+
+        # Converting to spark dataframe:
+        mappedTissues_spark = self.spark.createDataFrame(mappedTissues)
+
+        # Joining with cell lines:
+        return (
+            cell_df
+            .join(mappedTissues_spark, on='tissue', how='left')
+
+            # Generating the diseaseCellLines object:
+            .select('name', 'id', 'biomarkerList', struct(['tissue', 'name', 'id', 'tissueId']).alias('diseaseCellLines'))
```

Comment:
Yes, this is the case for all those evidence sources where the study was done on one single cell line. However for project score (crispr) this is a combination of experiments performed on a set of cell lines. It's ot ideal.

Reply:
I feel we are trying to grasp too much complexity... however I see the value of having this resolution: as different cell lines have different biomarker compositions, it would (one day) allow a more granular questions.

Reply:
Ah, okay, I see. Just one more question though, to confirm I understand this right: it's intended that in the schema this field will allow both a single ('tissue', 'name', 'id', 'tissueId') struct, and a list of such structs?

Reply:
But, and sorry if I'm being dense here, wouldn't `struct(['tissue', 'name', 'id', 'tissueId'])` just generate a single structured object rather than a list? I.e., what I think it will generate is this:

```
{'tissue': 'foo', 'name': 'bar', 'id': 123, 'tissueId': 456}
```

While the schema will expect:
```
[
    {'tissue': 'foo', 'name': 'bar', 'id': 123, 'tissueId': 456}
]
```

Reply:
You are right, the above expression generates a single struct. I made changes to reflect this.

Reply:
Later in the code there's this line:
```python
            merged_dataset.withColumn(
                "diseaseCellLines", f.array(f.col("diseaseCellLine"))
            )
```